### PR TITLE
Split JS Bundled Files, Update References

### LINKED ISSUE #346: Split apart JavaScript Bundle to improve loading experience
The JS bundle in GeoBlazor is currently 11GB for Core or 21GB for Pro.

This gissue is to explore and attempt to split the JS files for easier/faster loading.

### PR DESCRIPTION
**Pull request by TimPurdum**
Closes #346
 
While I don't see any immediate performance improvement by splitting the bundled JS, I think this is the right step forward, and makes future performance improvements possible.

In this PR
- Split bundled JS files
- Bumped all nuget and npm dependencies, including ArcGIS JS 4.30.9
- Added `MapView.PopupEnabled` and deprecated `PopupWidget.AutoOpenEnabled` to follow ArcGIS changes
- Added `ScaleUnit.Imperial` and deprecated `ScaleUnit.NonMetric` in `ScaleBarWidget` options to follow ArcGIS changes
- Added support for round-trip JSON serializing graphics using System.Text.Json.
- Implemented `IEquatable` and equality operators on `Dimension`
- Minor testing bug fixes in TypeScript

- Build Fixes
  - `assetCopy.ps1` - rather than continuing to fight powershell about the path, which sometimes produces `wwwroot/assets` and sometimes `wwwroot/assets/assets` based on environment running the script, I updated it to check for the duplicate folder and move everything at the end if it finds that.
  - Added environment variables to build test and sample projects with nuget packages or project references

### CODE REVIEW

File: src/dymaptic.GeoBlazor.Core/Objects/AttributesDictionary.cs [hunk 1]
```diff
@@ -372,9 +369,15 @@ internal class AttributesDictionaryConverter : JsonConverter<AttributesDictionar
     public override AttributesDictionary? Read(ref Utf8JsonReader reader, Type typeToConvert,
         JsonSerializerOptions options)
     {
+        // if first symbol is array
+        if (reader.TokenType == JsonTokenType.StartArray)
+        {
+            AttributeSerializationRecord[]? records = JsonSerializer.Deserialize<AttributeSerializationRecord[]>(ref reader, options);
```
Comment (AndersenBell) [inline src/dymaptic.GeoBlazor.Core/Objects/AttributesDictionary.cs:375]: nice, I didn't realzie there was an easy way to check if its an aeeay

**Review by AndersenBell — APPROVED** [review]

File: src/dymaptic.GeoBlazor.Core/Scripts/jsBuilder.ts [hunk 2]
```diff
@@ -1163,6 +1164,13 @@ export async function buildJsPopup(dotNetPopup: any, viewId: string): Promise<Po
         spinnerEnabled: dotNetPopup.spinnerEnabled ?? true
     });
     
+    if (hasValue(dotNetPopup.autoOpenEnabled)) {
```
Comment (seahro) [inline src/dymaptic.GeoBlazor.Core/Scripts/jsBuilder.ts:1167]: if `autoOpenEnabled` is listed as obsolete, do we want to update the `autoOpenEnabled` references and replace all references with popupEnabled?

File: samples/dymaptic.GeoBlazor.Core.Sample.Shared/Pages/QueryRelatedFeatures.razor [hunk 3]
```diff
@@ -13,7 +13,8 @@
 <MapView Class="map-view"
          OnClick="OnClick"
          Longitude="-98.5795" Latitude="39.8282"
-         Zoom="3">
+         Zoom="3"
+         PopupEnabled="false">
```
Comment (seahro) [inline samples/dymaptic.GeoBlazor.Core.Sample.Shared/Pages/QueryRelatedFeatures.razor:17]: I answered my own question on what is the default value when I read the Summary in `QueryRelatedFeatures`

**Review by seahro — APPROVED** [review]
very cool, one question about removing the "obsolete" reference

File: src/dymaptic.GeoBlazor.Core/Scripts/jsBuilder.ts (hunk 2 above)
Comment (TimPurdum) [reply src/dymaptic.GeoBlazor.Core/Scripts/jsBuilder.ts:1167]: In the samples? Yes, we should do that. I thought I had, but I will double check.

**Pull request merged**